lib/puppet/pops/validation/checker4_0.rb

```diff
@@ -623,9 +623,9 @@ def get_module_relative_path(file_path, modulepath_directories)
 
   def is_parent_dir_of(parent_dir, child_dir)
     parent_dir_path = Pathname.new(parent_dir)
-    clean_parent = parent_dir_path.cleanpath
+    clean_parent = parent_dir_path.cleanpath.to_s + File::SEPARATOR
 
-    return child_dir.to_s.start_with?(clean_parent.to_s)
+    return child_dir.to_s.start_with?(clean_parent)
```

Comment:
I verified `Pathname#cleanpath` normalizes backslashes to forward slashes on Windows, and strips trailing forward and backslashes, so it's safe to append `File::SEPARATOR` which is always forward slash.

```
irb(main):001:0> require 'pathname'
=> false
irb(main):002:0> Pathname.new('C:\foo\\').cleanpath.to_s
=> "C:/foo"
...
irb(main):008:0> Pathname.new('C:\foo\bar\\\\').cleanpath.to_s
=> "C:/foo/bar"
irb(main):009:0> Pathname.new('C:\foo\bar\\\\').cleanpath.to_s
=> "C:/foo/bar"
irb(main):010:0> File::SEPARATOR
=> "/"
irb(main):011:0> File::ALT_SEPARATOR
=> "\\"
...
irb(main):012:0> Pathname.new('C:/foo/bar/').cleanpath.to_s
=> "C:/foo/bar"
irb(main):013:0> Pathname.new('C:/foo/bar///').cleanpath.to_s
=> "C:/foo/bar"
```

Reply:
Thanks @joshcooper , I am still trying to get a windows test environment set up.